### ISSUE TITLE
Create a MonoPitchNet model prototype

### ISSUE THREAD

**Issue by anthonio9** [anthonio9/poly-pitch-net#3]
Create a model class - MonoPitchNet

* just a few CNN layers
* do not use reshape, use stack, chunk or permute
* 360 logits on the output, just like crepe
* take a lot of inspiration from penn

**Comment by anthonio9** [comment]
The model sketch is pretty simple, 3 conv2D layers, one conv1D at the end - a fully convolutional network.

**Comment by anthonio9** [comment]
MonoPitchNet is a simple model that as data accepts only one frequency transform block (CQT or STFT) per time step, let's say that for 5 time steps and 256 frequency bins the expected input should be `[5, 256]`,  if batched and batch size is 5, then `[5, 5, 256]`. The network will transpose the data into `[5, 256, 5]` in the forward function.

**Comment by anthonio9** [comment]
Reduction and maxpool2d removed from the model, because it's not even 2d tensors anymore. `input.permute(0, 2,1)` is used to transform the input tensor of shape `[B, T, C]` , `B` - batch size, `T` - time frames, `C` - frequency bins, into `[B, C, T]`.

**Comment by anthonio9** [comment]
It seems like the `MonoPitchNet1D` is in a good spot right now, *BCE* loss got down to 0.0033.
![image](https://github.com/anthonio9/poly-pitch-net/assets/66279891/bb5412b3-bfa8-4848-908d-e63d3d368802)

**Comment by anthonio9** [comment]
1-string plots are very far from ground truth, but at least is seems that the network has learned the frequencies related to the string in question - String nr 3.
![image](https://github.com/anthonio9/poly-pitch-net/assets/66279891/a63e4b31-0ac9-4c9a-aef1-fc2b188781d8)

**Comment by anthonio9** [comment]
Okay, let's call it a very bad prototype, now let's go to the next issue.
